tensorflow/python/distribute/input_lib.py

```diff
@@ -1700,6 +1777,22 @@ def initialize(self):
     return []
 
 
+def _create_iterators_per_replica_with_input_context(input_contexts,
+                                                    input_workers,
+                                                    dataset_fn):
+  """Create a multidevice iterator per workers given a dataset function."""
+  iterators = []
+  for i, ctx in enumerate(input_contexts):
+    devices = input_workers.compute_devices_for_worker(i)
+    with ops.device(devices[0]):
+      dataset = dataset_fn(ctx)
+      # Wrapping dataset here (ex. applying options) might result in moving it to the CPU
+      iterator = _SingleReplicaDatasetIterator(dataset, devices[0])
```

Comment:
i think here you should be passing `worker=input_workers.worker_devices[i], devices=input_workers.compute_devices_for_worker(i)`

Reply:
Please refer to https://github.com/tensorflow/tensorflow/pull/38968#discussion_r489125815